osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -363,6 +363,30 @@ public void Delete(Expression<Func<BeatmapSetInfo, bool>>? filter = null, bool s
             });
         }
 
+        /// <summary>
+        /// Delete a beatmap difficulty immediately.
+        /// </summary>
+        /// <remarks>
+        /// There's no undoing this operation, as we don't have a soft-deletion flag on <see cref="BeatmapInfo"/>.
+        /// This may be a future consideration if there's a user requirement for undeleting support.
+        /// </remarks>
+        public void DeleteDifficultyImmediately(BeatmapInfo beatmapInfo)
+        {
+            Realm.Write(r =>
+            {
+                if (!beatmapInfo.IsManaged)
+                    beatmapInfo = r.Find<BeatmapInfo>(beatmapInfo.ID);
+
+                Debug.Assert(beatmapInfo.BeatmapSet != null);
+                Debug.Assert(beatmapInfo.File != null);
+
+                var setInfo = beatmapInfo.BeatmapSet;
+
+                DeleteFile(setInfo, beatmapInfo.File);
+                setInfo.Beatmaps.Remove(beatmapInfo);
```

Comment:
This is missing updating beatmapset hash and changing status to `LocallyModified`, but not 100% sure if this should also invoke `ProcessBeatmap` as well.

Reply:
I don't think calling `ProcessBeatmap` is required, but I've added the hash change, good call.